token/js/src/instructions/initializePermanentDelegate.ts

```diff
@@ -0,0 +1,139 @@
+import { struct, u8 } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import type { AccountMeta } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import { TransactionInstruction } from '@solana/web3.js';
+import { programSupportsExtensions } from '../constants.js';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+    TokenUnsupportedInstructionError,
+} from '../errors.js';
+import { TokenInstruction } from './types.js';
+
+/** TODO: docs */
+export interface InitializePermanentDelegateInstructionData {
+    instruction: TokenInstruction.InitializePermanentDelegate;
+    delegate: PublicKey;
+}
+
+/** TODO: docs */
+export const initializePermanentDelegateInstructionData = struct<InitializePermanentDelegateInstructionData>([
+    u8('instruction'),
+    publicKey('delegate'),
+]);
+
+/**
+ * Construct an InitializePermanentDelegate instruction
+ *
+ * @param mint               Token mint account
+ * @param permanentDelegate  Authority that may sign for `Transfer`s and `Burn`s on any account
+ * @param programId          SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createInitializePermanentDelegateInstruction(
+    mint: PublicKey,
+    permanentDelegate: PublicKey | null,
+    programId: PublicKey
+): TransactionInstruction {
+    if (!programSupportsExtensions(programId)) {
+        throw new TokenUnsupportedInstructionError();
+    }
+    const keys = [{ pubkey: mint, isSigner: false, isWritable: true }];
+
+    const data = Buffer.alloc(initializePermanentDelegateInstructionData.span);
+    initializePermanentDelegateInstructionData.encode(
+        {
+            instruction: TokenInstruction.InitializePermanentDelegate,
+            delegate: permanentDelegate || new PublicKey(0),
+        },
+        data
+    );
+
+    return new TransactionInstruction({ keys, programId, data });
+}
+
+/** A decoded, valid InitializePermanentDelegate instruction */
+export interface DecodedInitializePermanentDelegateInstruction {
+    programId: PublicKey;
+    keys: {
+        mint: AccountMeta;
+    };
+    data: {
+        instruction: TokenInstruction.InitializePermanentDelegate;
+        delegate: PublicKey | null;
+    };
+}
+
+/**
+ * Decode an InitializePermanentDelegate instruction and validate it
+ *
+ * @param instruction Transaction instruction to decode
+ * @param programId   SPL Token program account
+ *
+ * @return Decoded, valid instruction
+ */
+export function decodedInitializePermanentDelegateInstruction(
```

Comment:
nit: for consistency
```suggestion
export function decodeInitializePermanentDelegateInstruction(
```